src/main/java/com/amihaiemil/zold/RtWallet.java

```diff
@@ -143,12 +144,12 @@ public void pay(
     }
 
     @Override
-    public JsonArray find(
-            final String id, final String details
-    ) throws IOException, URISyntaxException {
+    public JsonArray find(final JsonObject params)
```

Comment:
@ammaratef45 I think it would be better to use `Map<String, String>` here instead of `JsonObject`, because it will enforce `String`-type values and it has flat structure (without nested objects).